src/index.js

```diff
@@ -25,6 +25,20 @@ function getEventHandlerOptions(instance, eventName) {
   return handlerOptions;
 }
 
+// borrowed from react-is
+const hasSymbol = typeof Symbol === 'function' && Symbol.for;
+const REACT_FORWARD_REF_TYPE = hasSymbol ? Symbol.for('react.forward_ref') : 0xead0;
+
+/**
+ * Check if component is either a forwardRef or a class component
+ */
+function acceptsRef(WrappedComponent) {
+  return (
+    (WrappedComponent.$$typeof && WrappedComponent.$$typeof === REACT_FORWARD_REF_TYPE) ||
```

Comment:
I am unclear where this `$$type` is coming from, but I'd very much like to not use `$` in var names when the code already uses `_` to mark private functions and variables.

Reply:
`$$typeof` is an internal react property used to brand react elements and components. It is used by `react-is`, as well as by react internals.

Reply:
Is this publically documented? If not, let's not use it, because it cannot be relied on unless there's official documentation that says that's what it's called, and that's what it's for.

Reply:
Reading through https://reactjs.org/docs/forwarding-refs.html, it looks like the actual way to check is to, at run time, see if there's a `ref.current`. If there is, we're dealing with a forwarded ref. If there is not, we're dealing directly with a component.

Reply:
It is not publicly documented, but it is used in the separately published first party package react-is https://github.com/facebook/react/blob/master/packages/react-is

It is not an artifact of the build system, but it appears in the source code of react-is.
https://github.com/facebook/react/blob/master/packages/react-is/src/ReactIs.js#L30
https://github.com/facebook/react/blob/master/packages/shared/ReactSymbols.js

Also, react-is does not have any version constraints on what versions of react it is compatible with, which I can only assume means the brands are stable. https://github.com/facebook/react/blob/master/packages/react-is/package.json